openqabot/approver.py

```diff
@@ -37,58 +58,60 @@ def __init__(self, args: Namespace) -> None:
 
     def __call__(self) -> int:
-        logger.info("Start approving incidents in IBS")
+        log.info("Start approving incidents in IBS")
         increqs = (
             get_single_incident(self.token, self.single_incident)
             if self.single_incident
             else get_incidents_approver(self.token)
         )
 
         overall_result = True
-        incidents_to_approve = []
-
-        for inc in increqs:
-            try:
-                i_jobs = get_incident_settings(inc.inc, self.token, self.all_incidents)
-            except NoResultsError as e:
-                logger.info(e)
-                continue
-            try:
-                u_jobs = get_aggregate_settings(inc.inc, self.token)
-            except NoResultsError as e:
-                logger.info(e)
+        incidents_to_approve = [inc for inc in increqs if self._approvable(inc)]
 
-                if any(i.withAggregate for i in i_jobs):
-                    logger.info("Aggregate missing for %s" % str(inc.inc))
-                    continue
-
-                u_jobs = []
-
-            if not self.get_incident_result(i_jobs, "api/jobs/incident/", inc.inc):
-                logger.info("Inc %s has failed job in incidents" % str(inc.inc))
-                continue
-
-            if any(i.withAggregate for i in i_jobs):
-                if not self.get_incident_result(u_jobs, "api/jobs/update/", inc.inc):
-                    logger.info("Inc %s has failed job in aggregates" % str(inc.inc))
-                    continue
-
-            # everything is green --> approve inc
-            incidents_to_approve.append(inc)
-
-        logger.info("Incidents to approve:")
+        log.info("Incidents to approve:")
         for inc in incidents_to_approve:
-            logger.info(OBS_MAINT_PRJ + ":%s:%s" % (str(inc.inc), str(inc.req)))
+            log.info(OBS_MAINT_PRJ + ":%s:%s" % (str(inc.inc), str(inc.req)))
 
         if not self.dry:
             osc.conf.get_config(override_apiurl=OBS_URL)
             for inc in incidents_to_approve:
                 overall_result &= self.osc_approve(inc)
 
-        logger.info("End of bot run")
+        log.info("End of bot run")
 
         return 0 if overall_result else 1
 
+    def _approvable(self, inc: IncReq) -> bool:
+        try:
+            i_jobs = get_incident_settings(inc.inc, self.token, self.all_incidents)
+        except NoResultsError as e:
+            log.info(e)
+            return False
+        try:
+            u_jobs = get_aggregate_settings(inc.inc, self.token)
+        except NoResultsError as e:
+            log.info(e)
+
+            if any(i.withAggregate for i in i_jobs):
+                log.info("Aggregate missing for %s" % str(inc.inc))
+                return False
+
+            u_jobs = []
+
+        if not self.get_incident_result(i_jobs, "api/jobs/incident/", inc.inc):
+            log.info(
+                "Inc %s has at least one failed job in incident tests" % str(inc.inc)
```

Comment:
Why the "at least" here and not in line 109?

Reply:
I think I just missed that. Thx.

Created https://github.com/openSUSE/qem-bot/pull/87